### ISSUE TITLE
No test result output of running test after upgrading to 1.3.0 and newer

### ISSUE THREAD

**Issue by renegart** [litixsoft/karma-mocha-reporter#53]
I've used karma-mocha-reporter-1.1.5 for a long time. During an update of my system, I've upgraded also to the latest version. Since than I don't see any test results during running the tests.

I've played around with several reporter versions and it seem the "issue" was introduces with 1.3.0. Older versions are working, newer not. You have changed the behaviour of printout in 1.3.0 if multiple browsers are running the tests in parallel. I've only one browser which I connect manually to karma to run the tests. Seems there is something broken. Or do you have introduced a setting which may prevent the output?

**Comment by 4kochi** [comment]
The change in v1.3.0 was that the reporter waits that all browser have run a test before printing an output for the test. (https://github.com/litixsoft/karma-mocha-reporter/commit/3342eb8008c620c5b6146c899b45ef5d9559abbb) 

Maybe you have an example of your test so that I can reproduce the problem.

**Comment by renegart** [comment]
not sure how to give an example. Try to strip down the test suite.

What I do, I start karma on my developer machine and connect from another system with a chrome browser to the running karma instance. In my opinion nothing special :)

**Comment by tatablack** [comment]
Not sure whether I should open a new issue for this, but.. I also see no output with the latest version of this reporter (2.0.3) and Karma 0.13.20. _However_, If I downgrade Karma to 0.13.19 (or lower, I tried a few) then it works again.

Seems like some of their latest changes introduced a bug, or changed some internal behaviour.

**Comment by 4kochi** [comment]
Thanks for information @tatablack. I will check the problem with different karma versions.

**Comment by 4kochi** [comment]
I tested with some versions of karma below 0.13.20. But i still cannot reproduce the problem. Maybe you can send me a `karma.conf.js` file and the corresponding `spec` file. @renegart @tatablack

**Comment by tatablack** [comment]
Sorry, maybe I wasn't clear.. with versions of Karma _below_ 0.13.20 **it works**. With 0.13.20 or 0.13.22, for example, it doesn't. Let me know if you still cannot reproduce, I'll see if I can put something together.

**Comment by 4kochi** [comment]
I still cannot find an issue. It also works fine in travis (https://travis-ci.org/litixsoft/karma-mocha-reporter/jobs/128496141)

Maybe you can just clone this project locally and run `npm install` and then `npm test`. Just to see if this problem specific to your computer or project.

**Comment by 4kochi** [comment]
Hey @tatablack,

do you still have the problem?

**Comment by renegart** [comment]
Sorry, for not answering. I've overloaded with tasks at work at the moment. When I find some minutes next week, I'll send you a config file.

**Comment by renegart** [comment]
I've written a very small reproducer.
It contains a test file, a karma config and a text file which contains the expected output and the output which I get with karma-mocha-reporter@2.0.3

You'll see the test result are not printed, you see just the summary.

Steps to reproduce:
1) start karma on 'developer system'
2) start browser on 'test machine'
3) connect running karma server with 'http://<developer machine>:9876

[mocka_reporter.zip](https://github.com/litixsoft/karma-mocha-reporter/files/297081/mocka_reporter.zip)

**Comment by 4kochi** [comment]
Thank @renegart for your investigations. I was able to reproduce the problem with your files. Fix released as v2.0.4
